Crash when an envelope can't be found

I have an ESP32-S3 chipset. I am trying to install jaguar on the chipset but failing with the following error. 

```
❯ jag flash -p /dev/ttyACM1 -c ESP32-S3

Downloading ESP32-S3 firmware from https://github.com/toitlang/envelopes/releases/download/v2.0.0-alpha.160/firmware-ESP32-S3.envelope.gz ...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb2e6a8]

goroutine 1 [running]:
github.com/toitlang/jaguar/cmd/jag/commands.downloadGzipped({0xe089d8?, 0xc000199c50?}, {0xc000042a10?, 0xccf6a2?}, {0xc00007f3e0, 0x52})
	/home/runner/work/jaguar/jaguar/cmd/jag/commands/envelope.go:44 +0x48
github.com/toitlang/jaguar/cmd/jag/commands.downloadPublishedFirmware({0xe089d8, 0xc000199c50}, {0xc000040cc0, 0x10}, {0x7ffe28ed88b5, 0x8})
	/home/runner/work/jaguar/jaguar/cmd/jag/commands/envelope.go:136 +0x2ad
github.com/toitlang/jaguar/cmd/jag/commands.GetCachedFirmwareEnvelopePath({0xe089d8, 0xc000199c50}, {0xc000040cc0, 0x10}, {0x7ffe28ed88b5, 0x8})
	/home/runner/work/jaguar/jaguar/cmd/jag/commands/envelope.go:27 +0x105
github.com/toitlang/jaguar/cmd/jag/commands.FlashCmd.func1(0xc0001ecf08, {0xc0001ce840, 0x0, 0xccf6d2?})
	/home/runner/work/jaguar/jaguar/cmd/jag/commands/flash.go:97 +0x6e5
github.com/spf13/cobra.(*Command).execute(0xc0001ecf08, {0xc0001ce800, 0x4, 0x4})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983 +0xaca
github.com/spf13/cobra.(*Command).ExecuteC(0xc0001e2608)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/spf13/cobra.(*Command).ExecuteContext(0xcd1106?, {0xe089d8?, 0xc000199c50?})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1032 +0x47
main.main()
	/home/runner/work/jaguar/jaguar/cmd/jag/main.go:34 +0x18b
```

--- a/cmd/jag/commands/envelope.go
+++ b/cmd/jag/commands/envelope.go
@@ -18,6 +18,8 @@ import (
 // GetCachedFirmwareEnvelopePath returns the path to the cached firmware envelope.
 // If necessary, downloads the envelope from the server first.
 func GetCachedFirmwareEnvelopePath(ctx context.Context, version string, model string) (string, error) {
+	// Envelopes published in the Toit envelope repository are always lowercase.
+	model = strings.ToLower(model)
 	path, err := getFirmwareEnvelopePath(version, model)
 	if err != nil && err != os.ErrNotExist {
 		return "", err
@@ -25,6 +27,16 @@ func GetCachedFirmwareEnvelopePath(ctx context.Context, version string, model st
 	if err == os.ErrNotExist {
 		// Download the envelope from the server.
 		if err := downloadPublishedFirmware(ctx, version, model); err != nil {
+			fmt.Printf("Failed to download firmware: %v\n", err)
+			switch model {
+			case "esp32", "esp32s3", "esp32c3", "esp32s2":
+				// These names are correct. Simply return the error.
+			case "ESP32-S3", "ESP32-C3", "ESP32-S2":
+				fmt.Printf("Chip model names must be lowercase without dashes. Please try again with 'esp32s3', 'esp32c3', or 'esp32s2'.\n")
+			default:
+				fmt.Printf("Make sure the model name is correct. You can find supported models at\n")
+				fmt.Printf("https://github.com/toitlang/envelopes/releases/tag/%s.\n", version)
+			}
 			return "", err
 		}
 	}
@@ -41,11 +53,10 @@ func isURL(path string) bool {
 
 func downloadGzipped(ctx context.Context, url string, path string) error {
 	bundle, err := download(ctx, url)
-	defer bundle.Close()
 	if err != nil {
 		return err
 	}
-
+	defer bundle.Close()
 	return storeGzipped(bundle, path)
 }
 

